tools/bnfformatter.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "CLI11.hpp"
+
+class BnfFormatter : public CLI::Formatter {
+  public:
+    std::string make_option_opts(const CLI::Option* opt) const override;
+};
+
+std::string BnfFormatter::make_option_opts(const CLI::Option* opt) const {
```

Comment:
This code is mainly taken from https://github.com/AntelopeIO/antler-proj/blob/286ab12da71d3aeefdeb233f84a92ddf7a826bbd/tools/CLI11.hpp#L4093
changes are for handling `get_required` and `get_default_str`